include/casmutils/xtal/structure.hpp

```diff
@@ -48,29 +48,29 @@ class Structure
     template <typename CASMType> const CASMType& __get() const;
 
 private:
-    /// This function updates the appropriate members using
-    /// the flag given as a reference
-    void _update_using(std::string reference);
-    /// This function updates the casm_simplestructure member
+    ///  updates the appropriate members using
+    /// the argument given as a reference
+    template <typename StructureRepresentation> void _update_using(const StructureRepresentation& struc);
```

Comment:
You should choose between template specialization passing no arguments, or overload resolution passing the structure. Here you're mixing both. Based on how you've written other private members, I think it makes more sense to have template specialization with no arguments, since `struc` never gets used in the function anyway.